odxtools/communicationparameterref.py

```diff
@@ -41,7 +39,6 @@ def from_et(et_element: ElementTree.Element, doc_frags: List[OdxDocFragment],
         else:
             value = create_complex_value_from_et(odxrequire(et_element.find("COMPLEX-VALUE")))
 
-        is_functional = dl_type == DiagLayerType.FUNCTIONAL_GROUP
```

Comment:
This is the critical line of code where the parameter is set.

I used the DiagLayer's type to determine if a communication parameter is used for functional addressing or not. The reason for this heuristic is that all of the pdx files I worked with last year provided two almost indistinguishable communication parameters, one for functional addressing and one for non-functional addressing. Apart the surrounding DiagLayer and its type there is afair no other indication to what type of addressing the parameter refers to.

I suggest not to remove this heuristic unless a reasonable alternative is found, because some of our current projects use the query methods in the `DiagLayer` class, which depend on this parameter.

Reply:
the problem is that DiagServices may also contain referencces to communication parameters and this heuristic is not available because diagnostic services are inherited, i.e., does the commparam change its "allegiance" if it is inherited by an ECU-VARIANT that is derived from a FUNCTIONAL-GROUP?

Reply:
@andlaus the alternative to `is_functional` would be to expose to what `DiagLayerType` the comm param belongs to.

Reply:
> I suggest not to remove this heuristic unless a reasonable alternative is found, because some of our current projects use the query methods in the DiagLayer class, which depend on this parameter.

As this seems to be an issue with our PDX parameter naming conventions, I will forward this to my colleagues, who will eventually check if we can fix this internally without stopping the development of odxtools

Reply:
We have revisted our usage of this part of the API and found that there exists a fix for the behavior described above (at least) in the most recent version of odxtools. In case two communcation parameters have the same name, they can be distinguished by simply also providing the protocol name to the query.

The `is_functional` parameter / attribute is indeed no longer required and can be removed.